platform/posix/utest/reconnect_utest.c

```diff
@@ -0,0 +1,165 @@
+    TEST_ASSERT_GREATER_OR_EQUAL_UINT32( INITIAL_RECONNECT_BACKOFF_SECONDS,
+                                         reconnectParams.nextJitterMax );
+    TEST_ASSERT_LESS_THAN_UINT32( ( INITIAL_RECONNECT_BACKOFF_SECONDS +
+                                    MAX_JITTER_VALUE_SECONDS ),
+                                  reconnectParams.nextJitterMax );
+}
+
+/**
+ * @brief Test that #Transport_ReconnectParamsReset initializes the #nextJitterMax
+ * to be equal to some configurable backoff plus jitter.
+ */
+void test_Transport_ReconnectParamsReset_Sets_Jitter_Correctly( void )
+{
+    rand_ExpectAndReturn( RAND_RET_VAL );
+    Transport_ReconnectParamsReset( &reconnectParams );
+    verifyReconnectParamsAfterReset();
+}
+
+/**
+ * @brief Test that #Transport_ReconnectBackoffAndSleep is able to double the
+ * max value of the next jitter and return false when all attempts are exhausted.
+ *
+ * @note #Transport_ReconnectParamsReset is expected to be called before
+ * #Transport_ReconnectBackoffAndSleep.
+ */
+void test_Transport_ReconnectBackoffAndSleep_Succeeds( void )
+{
+    bool retriesArePending;
+    int32_t expectedNextJitterMax = INITIAL_RECONNECT_BACKOFF_SECONDS;
+    int32_t expectedAttemptsDone = 0;
+
+    rand_ExpectAndReturn( RAND_RET_VAL );
+    Transport_ReconnectParamsReset( &reconnectParams );
+    verifyReconnectParamsAfterReset();
+
+    expectedNextJitterMax = reconnectParams.nextJitterMax;
+
+    while( reconnectParams.attemptsDone < MAX_RECONNECT_ATTEMPTS )
+    {
+        expectedAttemptsDone++;
+
+        if( reconnectParams.nextJitterMax < ( MAX_RECONNECT_BACKOFF_SECONDS / 2U ) )
+        {
+            expectedNextJitterMax += expectedNextJitterMax;
+        }
+        else
+        {
+            expectedNextJitterMax = MAX_RECONNECT_BACKOFF_SECONDS;
+        }
+
+        rand_ExpectAndReturn( RAND_RET_VAL );
+        sleep_ExpectAndReturn( RAND_RET_VAL % expectedNextJitterMax, 0 );
+        retriesArePending = Transport_ReconnectBackoffAndSleep( &reconnectParams );
+        TEST_ASSERT_TRUE( retriesArePending );
+        TEST_ASSERT_EQUAL( expectedNextJitterMax, reconnectParams.nextJitterMax );
+        TEST_ASSERT_EQUAL( expectedAttemptsDone, reconnectParams.attemptsDone );
+    }
+
+    /* The next call to the function under test should now return that all
+     * attempts are now exhausted. */
+    rand_ExpectAndReturn( RAND_RET_VAL );
+    retriesArePending = Transport_ReconnectBackoffAndSleep( &reconnectParams );
+    TEST_ASSERT_FALSE( retriesArePending );
+
+    /* #Transport_ReconnectParamsReset is expected to be called once all attempts
+     * are exhausted. */
+    verifyReconnectParamsAfterReset();
+}
+
+/**
+ * @brief Test that the value of the next max jitter has a lower bound that will
+ * then be capped at some max threshold.
+ */
+void test_Transport_ReconnectBackoffAndSleep_Lower_Bound_Jitter_To_Cap( void )
+{
+    bool retriesArePending;
```

Comment:
(Nit) This could be a global and reset to false in `setUp`

Reply:
Similarly, `reconnectParams.attemptsDone = 0U;` initialization can be done in `setUp`